src/libcollections/str.rs

```diff
@@ -938,32 +1118,38 @@ impl str {
         core_str::StrExt::rsplit(self, pat)
     }
 
-    /// An iterator over substrings of `self`, separated by characters
-    /// matched by a pattern.
+    /// An iterator over substrings of the given string slice, separated by
+    /// characters matched by a pattern.
     ///
-    /// The pattern can be a simple `&str`, `char`, or a closure that
-    /// determines the split.
-    /// Additional libraries might provide more complex patterns
-    /// like regular expressions.
+    /// The pattern can be a `&str`, [`char`], or a closure that determines the
+    /// split.
     ///
-    /// Equivalent to `split`, except that the trailing substring
+    /// Equivalent to [`split()`], except that the trailing substring
```

Comment:
It is not equivalent then, is it? Maybe something like “It differs from [`split()`] in such and such ways” would be better?